### ISSUE TITLE
bad-version-info-comparison (PYI006) doesn't run in `.py` files

### ISSUE THREAD

**Issue by Avasam** [astral-sh/ruff#13836]
This rule could've caught an issue in pywin32 (https://github.com/mhammond/pywin32/pull/2416). I think it's a valid rule to enforce on runtime files to prevent incorrect expectations.

MRE:
```python
if sys.version_info > (3, 10):  # PYI006 only shows here in .pyi files
    print("Do something on Python 3.11")
```

Command: `ruff check --select=PYI`
Version: ruff 0.7.0

**Comment by AlexWaygood** [comment]
This makes sense to me.
